src/RandomForestBase.js

```diff
@@ -113,15 +113,16 @@ export default class RandomForestBase {
      */
     predict(toPredict) {
         var predictionValues = new Array(this.nEstimators);
+        toPredict = Matrix.checkMatrix(toPredict);
         for (var i = 0; i < this.nEstimators; ++i) {
-            var X = Utils.retrieveFeatures(new Matrix(toPredict), this.indexes[i]);
+            var X = toPredict.columnSelectionView(this.indexes[i]); // get features for estimator
             predictionValues[i] = this.estimators[i].predict(X);
         }
 
-        predictionValues = new Matrix(predictionValues).transpose();
-        var predictions = new Array(predictionValues.length);
-        for (i = 0; i < predictionValues.length; ++i) {
-            predictions[i] = this.selection(predictionValues[i]);
+        predictionValues = new Matrix(predictionValues).transposeView();
```

Comment:
I think that I cannot use that class because predictionValues is a 2D matrix, the other one is to use WrapMatrix2D but is the same as Matrix, I have to use tranposeView